client/client_test.go

```diff
@@ -196,3 +196,26 @@ func TestRetries(t *testing.T) {
 		})
 	}
 }
+
+func TestCancelable(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	server := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, r *http.Request) {
+			<-time.After(50 * time.Millisecond)
+			w.WriteHeader(http.StatusInternalServerError)
+		}))
+
+	ch := make(chan error)
+	go func() {
+		cli, _ := client.New()
+		_, err := cli.Put(ctx, server.URL)
+		ch <- err
+	}()
+
+	cancel()
+	err := <-ch
```

Comment:
This will hang if there is a bug. Don't you think we need to wrap it in a select with some timeout?

Reply:
[tests automatically timeout after a while](https://golang.org/cmd/go/#hdr-Testing_flags)

```
-timeout d
    If a test binary runs longer than duration d, panic.
    If d is 0, the timeout is disabled.
    The default is 10 minutes (10m).
```

Reply:
Are you saying that this is fine to wait 10m for a test to fail? 😀

Reply:
I think an easy fix would be to Ctrl-C the test :)

Keep in mind this is a library, there are no Rakefile to build nor to test. Test are only done in the IDE if the developer implementing a feature think about it.